### DISCUSSION TITLE
dbx sync dbfs to user workspace

### DISCUSSION THREAD

**Discussion by JPMarais-Inscorep** [databrickslabs/dbx#486] [Q&A] [answered]
When using `dbx sync repo --source .` to sync a local repo with a repo located in a specific user folder within the Repos area in Databricks, everything works as expected. This does however bring in something that one just needs to watch out for as the repo in Databricks will note that changes have been made to the repo, however, if these changes have already been pushed to the remote repo from the local environment, then these changes need to be discarded before a pull can be done on the repo within Databricks. Although this is not a problem per se, it is something to look out for.

In order to get around this, I thought one would be able to use the **dbfs** sync to rather sync the local repo to a folder in the user's workspace. Thus leaving the repo to be used as normal. This is especially helpful if you are busy on a new feature in a repo, but then might need to review another feature also in the same repo. The folder in the Workspace can be synced to the local changes for the new feature, and the repo can be used to review the other feature.

However, when syncing using the **dbfs** option, the local code is always synced the dbfs file system and not the user's workspace. Am I misunderstanding the use the **dbfs** sync option, or is there an option to use this option to sync to a folder in the user's workspace?

* dbx version used: 0.7.5
* Databricks Runtime version: 11.2

**Comment by janosroden** [comment, marked as answer]
You can't open a notebook from dbfs but you can import code from there.

The original PR explains how:
```python
import sys

if "/tmp/users/first.last/my-repo" not in sys.path:
  sys.path.insert(0, "/tmp/users/first.last/my-repo")
```

See https://github.com/databrickslabs/dbx/pull/239 for more details.
